lmdeploy/vl/model/minicpmv.py

```diff
@@ -54,6 +54,20 @@ def build_model(self):
         else:
             self._forward_func = self._forward_v2
 
+        # adapt new code commit 287e3f85
+        if not hasattr(model, 'slice_image'):
+            from transformers import AutoProcessor
+            processor = AutoProcessor.from_pretrained(self.model_path,
+                                                      trust_remote_code=True)
+            model.slice_image = processor.image_processor.slice_image
+
+            def _reshape_by_patch(x):
+                x = x.numpy()
```

Comment:
if x is on the device, should it be `x.cpu().numpy()`?

Reply:
I think we don't need to make it too complicated and follow the current upstream implemention is enough.

Currently, x(slice_image) is cpu tensor. 
https://huggingface.co/openbmb/MiniCPM-Llama3-V-2_5/blob/main/modeling_minicpmv.py#L53-L61

Reply:
Personally, it's safe to do so.

Reply:
I agree with @RunningLeon

Using `x.cpu().numpy()` is not too complicated. Moreover, it is safe